File: CPP_Graph/Graph.cpp
#include <iostream>
#include <list>
using namespace std;

class Graph
{
	int V; // number of  vertices
	list<int> *adj;
	void DFSrecur(int v, bool visited[]);
public:
	Graph(int V); //Constructor
	void addEdge(int v, int w);
	void DFScall(int v);
}

Graph :: Graph(int V)
{
	this->V = V;
	adj = new list<int>[V];
}

void Graph::addEdge(int v, int w)
{
	adj[v].push_back(w);
}

void Graph::DFSrecur(int v, bool visited[])
{
	visited[v] = true;
	cout << v << ' ';
	list<int>::iterator i;
	for(i = adj[v].begin() ; i != adj[v].end() ; ++i){
		if(visited[v] != true)
			DFSrecur(*i, visited);
	}
}

void Graph::DFScall(int V)
{
	bool *visited = new bool[V];
	for(int i = 0 ; i < V ; i++)
		visited[i] = false;
	DFSrecur(V, visited);
}

File: Python_Graph/Graph.py
from collections import defaultdict

class Graph:
	def __init__(self):
		self.graph = defaultdict(list)

	def addEdge(self, u, v):
		self.graph[u].append(v)
		self.graph[v].append(u)

	def printGraph(self):
		for u in self.graph:
			print(self.graph[u])

	def DFSrecur(self, v, visited):
		visited[v] = True
		print(v, ' ', end="")
		for u in self.graph[v]:
			if visited[u] != True:
				self.DFSrecur(u, visited)
	def DFScall(self,v):
		visited = [False]*(len(self.graph)+1)
		self.DFSrecur(v, visited)
		print()

	def BFScall(self, v):
		visited = [False]*(len(self.graph)+1)
		self.path = []
		q = []; q.append(v); visited[v] = True

		while q:
			v = q.pop(0)
			self.path.append(v)
			for adj in self.graph[v]:
				if visited[adj] == False:
					visited[adj] = True
					q.append(adj)
					



'''
	def DFSutil(self, v, visited):
		visited[v] = True
		print(v,)
		for i in self.graph[v]:
			if visited[i] == False:
				self.DFSutil(i, visited)

	def DFS(self,v):
		visited = [False]*(len(self.graph)+1)
		self.DFSutil(v,visited)	

	def BFS:
	'''